src/pages/Game.js

```diff
@@ -208,6 +208,17 @@ export default class Game extends Component {
     return lastMessage > this.state.lastReadChat;
   }
 
+  get userColorKey() {
+    return `user_color_${window.location.href}`;
```

Comment:
Hmm, I'm a bit curious how this part works. I _think_ user id is persistent across refreshes for any browser that has localStorage enabled, so shouldn't that guarantee that user colors are persistent as well?

Reply:
Though I'm now thinking that if userColorKey were a constant, then that would allow users to have a consistent color across games. Eventually i want to build a proper user system w/ persistent settings, profile (name & color), etc., but in the meantime having persistent color would be cool!

Reply:
So the problem that I was using local storage for is that we initially do not have all of the events reduced when we first load the game, so the color will be null and we will end up with a random color (and even worse changing our color to that, which means we lose whatever we would have reduced to).

To get around needing to understand and refactor the color state, I used the localstorage as a hammer so that we would at least have the color we think we should when we reload.


I can move us to storing the color directly so that the color is persisted throughout. I agree that moving to a more persistent profile system will certianly make this easier and probably would give a nice cleanup of this code.

Reply:
Ah I see, that makes sense! I forgot that a `setColor` event was sent on client-load (what a hack 😅), that's definitely a good reason to use localStorage to workaround.